vpr/src/route/router_lookahead.cpp

```diff
@@ -61,13 +61,19 @@ std::unique_ptr<RouterLookahead> make_router_lookahead(const t_det_routing_arch&
     return router_lookahead;
 }
 
+void RouterLookahead::scale_delay_and_cong_by_criticality(float& delay, float& cong, const t_conn_cost_params& params) const {
+    delay *= params.criticality;
+    cong *= 1. - params.criticality;
+}
```

Comment:
is this member functon is always called after `get_expected_delay_and_cong()`?
In this case, I think it is better to replace this member function with another one that computes the scaled congestion and delay

Reply:
Always except for the lookahead profiler. I moved the code in ``get_expected_delay_and_cong()`` into a new function ``get_expected_delay_and_cong_ignore_criticality()``, and now ``get_expected_delay_and_cong()`` just calls that new function + the scaling function. So, all the previous calls to ``get_expected_delay_and_cong()`` remain unmodified and the profiler can call ``get_expected_delay_and_cong_ignore_criticality()``.